src/mngt/ocf_mngt_cache.c

```diff
@@ -215,7 +217,9 @@ static ocf_error_t __init_cleaning_policy(ocf_cache_t cache)
 
 static void __deinit_cleaning_policy(ocf_cache_t cache)
 {
+	ocf_refcnt_freeze(&cache->cleaner.cleaning_ctx);
 	ocf_cleaning_deinitialize(cache);
+	ocf_refcnt_unfreeze(&cache->cleaner.cleaning_ctx);
```

Comment:
At this point there will be neither io in flight nor another mngt operation, so we can safely skip this.